lib/SIL/SILArgument.cpp

```diff
@@ -26,12 +26,6 @@ using namespace swift;
 SILArgument::SILArgument(SILBasicBlock *ParentBB, SILType Ty,
                          const ValueDecl *D)
   : ValueBase(ValueKind::SILArgument, Ty), ParentBB(ParentBB), Decl(D) {
-  // Function arguments need to have a decl.
```

Comment:
Hi @jckarter, if you have a better idea here, let me know. I'm not a fan of removing assertions for all the usual reasons.